MapboxAndroidDemo/proguard-rules.pro

```diff
@@ -116,18 +116,21 @@
 -dontwarn com.cunoraz.gifview.library.**
 
 # Mapbox
--keep class com.mapbox.android.telemetry.**
--keep class com.mapbox.android.core.location.**
--keep class android.arch.lifecycle.** { *; }
--keep class com.mapbox.android.core.location.** { *; }
--keep class com.mapbox.mapboxsdk.** { *; }
+-dontnote class com.mapbox.android.telemetry.**
```

Comment:
What was the reason to keep this and below classes before? Is it not needed anymore? What are the notes produced in this case?

Reply:
> What are the notes produced in this case?

See the screenshots above. 

> What was the reason to keep this and below classes before? Is it not needed anymore?

Was keeping them to cut down on Proguard messages, but the ones in the screenshots above were new. Based on https://github.com/mapbox/mapbox-android-demo/pull/1174#discussion_r318460372, sounded like switching things to `dontnote` was the remedy.

If you had to address the lines seen in say, https://user-images.githubusercontent.com/4394910/63796375-cd93d280-c8ba-11e9-9617-11c7b9c45fe4.png, what would you do?

Reply:
I agree with `dontnote` for gestures, plugins and maps. However, there's nothing about telemetry, location or lifecycle in those screenshots.

Reply:
ah, I see. Well I suspect they were there from past Proguard issues. You good with keeping `-keep` next to them and only doing `dontnote` for gestures, plugins, and maps?

Reply:
Sure, we can keep them unchanged and tackle in a separate PR if they are actually obsolete.

Reply:
Ok, adjusted them and re-based. Take a look when you get a chance 🙇